sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -198,7 +198,7 @@ function buildRoutePath(
       value = encodeURIComponent(value);
     }
 
-    routePath = routePath.replace(/\{\w+\}/, String(value));
+    routePath = routePath.replace(/\{[\w-]+\}/, String(value));
```

Comment:
Alternatively if the goal of the original PR https://github.com/Azure/azure-sdk-for-js/pull/24997 to allow anything but still handle multiple separators maybe a regex like `/\{[^}]+\}/` could be wider, but it may not be what we want https://regex101.com/r/kaRByL/1

Reply:
Happy either way personally. I can't think of any other chars that we might want to accept here? Up to you

Reply:
Good Q! underscores should be covered under the `\w` umbrella: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_class_escape#w

`\w` is basically equivalent to `[a-zA-Z0-9_]`

If you guys want, we can be more explicit in the regex: `/\{[a-zA-Z0-9_-]+\}/ or even use `/\{[^}]+\}/`

Knowing what the spec expects would help me decide, but barring that, we can just use what the PR has here